h2gis-utilities/src/main/java/org/h2gis/utilities/jts_utils/GeographyUtils.java

```diff
@@ -0,0 +1,128 @@
+     * @param distance expressed in degrees 
+     * @return a new envelope
+     */
+    public static Envelope expandEnvelopeByMeters(Envelope envelope, double distance) {
+        if (envelope.isNull()) {
+            return null;
+        }
+        if(distance==0){
+            return envelope;
+        }
+        else if(distance<0){
+            throw new IllegalArgumentException("Extend operation does not accept negative value");
+        }
+        
+        double minLon = envelope.getMinX();
+        double maxLon = envelope.getMaxX();
+        double maxLat = envelope.getMaxY();
+        double minLat = envelope.getMinY();
+        
+        //Check if the envelope has latitude, longitude co-ordinates        
+        if (!UTMUtils.isValidLatitude((float) minLat)){
+            throw new IllegalArgumentException("Invalid min latitude");
+        }
+        if (!UTMUtils.isValidLatitude((float) maxLat)){
+            throw new IllegalArgumentException("Invalid max latitude");
+        }
+        if (!UTMUtils.isValidLongitude((float) minLon)){
+            throw new IllegalArgumentException("Invalid min longitude");
+        }
+        if (!UTMUtils.isValidLongitude((float) maxLon)){
+            throw new IllegalArgumentException("Invalid max longitude");
+        }
+        double verticalExpansion = computeLatitudeDistance(distance);
+        double horizontalExpansion = computeLongitudeDistance(distance, Math.max(Math.abs(minLat), Math.abs(maxLat)));
+        
+        return new Envelope(Math.max(-180, minLon - horizontalExpansion), Math.min(180, maxLon + horizontalExpansion), 
+                Math.max(-90, minLat - verticalExpansion), Math.min(90, maxLat + verticalExpansion));
+    }   
+    
+    
+     /**
+     * Computes the amount of degrees of latitude for a given distance in meters.
+     *
+     * @param meters distance in meters
+     * @return latitude degrees
+     */
+    public static double computeLatitudeDistance(double meters) {
+        return (meters * 360) / (2 * Math.PI * RADIUS_OF_EARTH_IN_METERS);
+    }
+
+    /**
+     * Computes the amount of degrees of longitude for a given distance in meters.
+     *
+     * @param meters   distance in meters
+     * @param latitude the latitude at which the calculation should be performed
+     * @return longitude degrees
+     */
+    public static double computeLongitudeDistance(double meters, double latitude) {
+        return (meters * 360) / (2 * Math.PI * RADIUS_OF_EARTH_IN_METERS * Math.cos(Math.toRadians(latitude)));
+    }
+    
+    /**
+     * Calculate the spherical distance between two coordinates in meters using the
+     * Haversine formula. See https://fr.wikipedia.org/wiki/Formule_de_haversine
+     * This calculation is done using the approximate earth radius
+     *
+     * @param coordA
+     * @param coordB
+     * @return distance in meters
+     */
+    public static double getHaversineDistanceInMeters(Coordinate coordA, Coordinate coordB) {
```

Comment:
Bad coordinated (such as null or invalid) not managed.

Reply:
done